qdax/core/emitters/mees_emitter.py

```diff
@@ -0,0 +1,613 @@
+    get_cells_indices,
+)
+from qdax.core.emitters.emitter import Emitter, EmitterState
+from qdax.types import Descriptor, ExtraScores, Fitness, Genotype, RNGKey
+
+
+@dataclass
+class MEESConfig:
+    """Configuration for the MAP-Elites-ES emitter."""
+
+    sample_number: int = 1000  # num of samples for gradient estimate
+    sample_sigma: float = 0.02
+    sample_mirror: bool = True  # if 1, use mirroring sampling
+    sample_rank_norm: bool = True  # if 1, use normalisation
+    num_optimizer_steps: int = 10  # frequency of archive-sampling
+    adam_optimizer: bool = True  # if 1 use ADAM, if 0 use SGD
+    learning_rate: float = 0.01
+    learning_rate_decay: float = 1.0  # only applied when no ADAM
+    l2_coefficient: float = 0.02  # coefficient for regularisation
+    novelty_nearest_neighbors: int = 10
+    use_explore: bool = True  # if 0, use only fitness gradient
+
+
+class MEESEmitterState(EmitterState):
+    """Emitter State for the MAP-Elites-ES emitter."""
+
+    learning_rate: float
+    initial_optimizer_state: optax.OptState  # only used for ADAM
+    optimizer_state: optax.OptState  # only used for ADAM
+    gradient_offspring: Genotype  # Offspring generated through gradient
+    generation_count: int  # Generation counter
+    novelty_archive: Descriptor  # Used to compute novelty for explore
+    last_updated_genotypes: Genotype  # Used to sample parents
+    last_updated_fitnesses: Fitness
+    last_updated_position: jnp.array
+    random_key: RNGKey
+
+
+class MEESEmitter(Emitter):
+    """
+    Emitter reproducing the MAP-Elites-ES exploit-explore and MAP-Elites-ES exploit
+    algorithms from "Scaling MAP-Elites to Deep Neuroevolution" by Colas et al:
+    https://dl.acm.org/doi/pdf/10.1145/3377930.3390217
+
+    One can choose between the two algorithms by setting the use_explore param:
+        ME-ES exploit-explore: alternates between num_optimizer_steps of fitness
+            gradients and num_optimizer_steps of novelty gradients.
+        ME-ES eploit: only uses fitness gradient and no novelty gradients, resample
+            parent from the archive every num_optimizer_steps steps.
+    """
+
+    def __init__(
+        self,
+        config: MEESConfig,
+        total_generations: int,
+        scoring_fn: Callable[
+            [Genotype, RNGKey], Tuple[Fitness, Descriptor, ExtraScores, RNGKey]
+        ],
+        num_descriptors: int,
+    ) -> None:
+        """Initialise the MAP-Elites-ES emitter.
+
+        WARNING: total_generations is required to build the novelty archive.
+
+        Args:
+            config
+            total_generations: total number of generations for which the
+                emitter will run, allow to initialise the novelty archive.
+            scoring_fn: used to evaluate the samples for the gradient estimate.
+            num_descriptors: dimension of the descriptors, used to initialise
+                the empty novelty archive.
+
+        Returns: /
```

Comment:
I think that you can just remove the Returns line when it's void

Reply:
Done.